src/HOCON/Impl/HoconTokenizer.cs

```diff
@@ -221,698 +185,822 @@ public string GetHelpTextAtIndex(int index, int length=0)
         /// </summary>
-        /// <returns>A <see cref="TokenType.ArrayEnd"/> token from the tokenizer's current position.</returns>
-        public Token PullArrayEnd()
+        /// <returns>A <see cref="TokenType.StartOfArray"/> token from the tokenizer's current position.</returns>
+        private bool PullArrayStart(HoconTokenizerResult tokens)
         {
-            int start = Index;
-            if (!IsArrayEnd())
-            {
-                throw new HoconTokenizerException(string.Format("Expected end of array {0}", GetHelpTextAtIndex(start)));
-            }
+            if (!Matches('['))
+                return false;
+
             Take();
-            IsParsingArray = false;
-            return new Token(TokenType.ArrayEnd,start,Index-start);
+            tokens.Add(new Token("[", TokenType.StartOfArray, this));
+            return true;
         }
 
-        /// <summary>
-        /// Determines whether the current token matches an <see cref="TokenType.ArrayEnd"/> token.
-        /// </summary>
-        /// <returns><c>true</c> if the token matches; otherwise, <c>false</c>.</returns>
-        public bool IsArrayEnd()
+        private bool PullParenthesisStart(HoconTokenizerResult tokens)
         {
-            return Matches("]");
-        }
+            if (!Matches('('))
+                return false;
 
-        /// <summary>
-        /// Determines whether the current token matches an <see cref="TokenType.ArrayStart"/> token.
-        /// </summary>
-        /// <returns><c>true</c> if the token matches; otherwise, <c>false</c>.</returns>
-        public bool IsArrayStart()
-        {
-            return Matches("[");
+            Take();
+            tokens.Add(new Token("(", TokenType.ParenthesisStart, this));
+            return true;
         }
 
-        /// <summary>
-        /// Retrieves a <see cref="TokenType.ArrayStart"/> token from the tokenizer's current position.
-        /// </summary>
-        /// <returns>A <see cref="TokenType.ArrayStart"/> token from the tokenizer's current position.</returns>
-        public Token PullArrayStart()
+        private bool PullParenthesisEnd(HoconTokenizerResult tokens)
         {
-            int start = Index;
+            if (!Matches(')'))
+                return false;
+
             Take();
-            IsParsingArray = true;
-            return new Token(TokenType.ArrayStart,Index, Index-start);
+            tokens.Add(new Token(")", TokenType.ParenthesisEnd, this));
+            return true;
         }
 
-        /// <summary>
-        /// Retrieves a <see cref="TokenType.Dot"/> token from the tokenizer's current position.
-        /// </summary>
-        /// <returns>A <see cref="TokenType.Dot"/> token from the tokenizer's current position.</returns>
-        public Token PullDot()
+        private bool PullNewLine(HoconTokenizerResult tokens)
         {
-            int start = Index;
+            if (!Matches(Utils.NewLine))
```

Comment:
If you do a check for both cases (`\n` and `\r\n`), you should not need that replacement of Unix style endings at the intial HoconParser.Parse.

Reply:
Its already handled, `\n` and `\r\n` shouldn't matter anymore.